Sources/AEPOptimize/Optimize.swift

```diff
@@ -139,6 +128,43 @@ public class Optimize: NSObject, Extension {
         return true
     }
 
+    /// Processes the propositions request event, dispatched with type `EventType.optimize` and source `EventSource.requestContent`.
+    ///
+    /// It processes events based on the type of the requested event type
+    /// - Parameter event: propositions request event
+    private func processOptimizeRequestContent(event: Event) {
+        if event.isUpdateEvent {
+            processUpdatePropositions(event: event)
+        } else if event.isGetEvent {
+            guard let eventDecisionScopes: [DecisionScope] = event.getTypedData(for: OptimizeConstants.EventDataKeys.DECISION_SCOPES),
+                  !eventDecisionScopes.isEmpty
+            else {
+                /// If decision scopes are not present in the event data, then adding it to the event queue
+                eventsQueue.add(event)
```

Comment:
We should never run into this scenario because its validated in the public API, but if decision scope list in the event data is empty, can we not queue the event and dispatch error response instead? Queuing it will eventually cause it to fail when its processed in `processGetPropositions`. I propose moving the below logic here and failing fast. Thoughts?
https://github.com/adobe/aepsdk-optimize-ios/blob/main/Sources/AEPOptimize/Optimize.swift#L365-L371